third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -536,10 +552,12 @@ impl ValueImpl {
             (Bool(l), Bool(r)) => l == r,
             (Address(l), Address(r)) => l == r,
 
-            (Container(l), Container(r)) => l.equals(r)?,
+            (Container(l), Container(r)) => l.equals(r, depth, max_depth)?,
 
-            (ContainerRef(l), ContainerRef(r)) => l.equals(r)?,
-            (IndexedRef(l), IndexedRef(r)) => l.equals(r)?,
+            // Unlike for container, increment depths here because we cover for the containers the
```

Comment:
This comment is confusing. Do you want to say we count the reference as one nesting?